server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/builder/OnHeapStarTreeBuilder.java

```diff
@@ -271,18 +274,16 @@ public StarTreeDocument next() {
      */
     private void sortStarTreeDocumentsFromDimensionId(StarTreeDocument[] starTreeDocuments, int dimensionId) {
         Arrays.sort(starTreeDocuments, (o1, o2) -> {
+            List<Dimension> dimensionsOrder = starTreeField.getDimensionsOrder();
             for (int i = dimensionId; i < numDimensions; i++) {
                 if (!Objects.equals(o1.dimensions[i], o2.dimensions[i])) {
-                    if (o1.dimensions[i] == null && o2.dimensions[i] == null) {
-                        return 0;
+                    Dimension dimension = dimensionsOrder.get(i);
+                    if (dimension instanceof UnsignedLongDimension) {
```

Comment:
why do we need this check ? I'm assuming line no 285, 286 is suffice.

Reply:
Yes - Removed.